lib/Conversion/TorchToLinalg/Reduction.cpp

```diff
@@ -551,6 +568,11 @@ class ConvertReductionOp : public ConversionPattern {
     if (isa<AtenAllDimOp>(op) && elemType.isa<mlir::IntegerType>() &&
         elemType.getIntOrFloatBitWidth() == 8)
       return rewriter.notifyMatchFailure(op, "uint8 is not supported");
+
+    if (isa<AtenNormScalarOp>(op) &&
+        !(elemType.isa<mlir::FloatType>() || elemType.isa<mlir::ComplexType>()))
+      return rewriter.notifyMatchFailure(
+          op, "only float and complex types are valid for norm op");
```

Comment:
Is this a constraint from pytorch? Please include a comment in the code, if so.  You could use an op verifier for this (see for example Torch_AtenPermuteOp)